src/Compilers/CSharp/Portable/Binder/SubsumptionDiagnosticBuilder.cs

```diff
@@ -129,6 +129,8 @@ private ErrorCode CheckSubsumed(BoundPattern pattern, DecisionTree decisionTree,
                         // the input can no longer be null. In that case a null pattern is subsumed.
                         if (isNull && !inputCouldBeNull)
                         {
+                            // Note: we do not have any test covering this. Is it reachable?
```

Comment:
I have not attempted to produce a proof that the code is unreachable, which would be a prerequisite for throwing there.